node/rpcstack_test.go

```diff
@@ -54,25 +54,111 @@ func TestVhosts(t *testing.T) {
 
 // TestWebsocketOrigins makes sure the websocket origins are properly handled on the websocket server.
 func TestWebsocketOrigins(t *testing.T) {
+	tryWebsocketOriginsSimpleRule(t)
+	// tryWebsocketOriginsRuleWithScheme(t)
+	// tryWebsocketOriginsRuleWithPort(t)
+	// tryWebsocketOriginsRuleWithSchemeAndPort(t)
+}
+
+func tryWebsocketOriginsSimpleRule(t *testing.T) {
 	srv := createAndStartServer(t, httpConfig{}, true, wsConfig{Origins: []string{"test"}})
 	defer srv.stop()
 
-	dialer := websocket.DefaultDialer
-	_, _, err := dialer.Dial("ws://"+srv.listenAddr(), http.Header{
-		"Content-type":          []string{"application/json"},
-		"Sec-WebSocket-Version": []string{"13"},
-		"Origin":                []string{"test"},
-	})
-	assert.NoError(t, err)
+	assert.NoError(t, attemptWebsocketConnectionFromOrigin(t, srv, "test"))
```

Comment:
There is no need to handle all kinds of values for the Origin header because it will be either "null" or a fully qualified URL. See [RFC6454, Section 7.1](https://tools.ietf.org/html/rfc6454#section-7.1) for the formal
definition of Origin.

Reply:
Formally, the origin can also be a comma-separated list of origin URLs, but we only need to handle the first one. Not sure what to do about "null", maybe we should just always reject it or have a special case for it.

Reply:
The formal test that the code has to pass uses just "test" instead of a fully qualified URL, and there is a sufficient amount of documentation out there suggesting that "localhost" suffices as a command-line parameter.  While I also thought that the Origin from a browser would only be a fully qualified URL, the test suite disagreed with me so I wrote the program in a way that would pass the tests.

Reply:
Well, the test was wrong then. I appreciate your work on this, let's just get it right so it will really work with browsers.